### DISCUSSION TITLE
[Proposal] HttpClient.Shared

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#65261] [Ideas]
Just like Random.Shared, it would be nice to have HttpClient.Shared.
Since initializing HttpClient is such a bad idea until you have to do it, would be nice to just use a shared static instance for simple api calls instead.
Another feature could include static HttpClient methods that internally uses HttpClient.Shared so we don't even have to type this in code

**Comment by EgorBo** [comment]
`Random.Shared` returns a sort of immutable data while HttpClient.Shared can be re-configured in different places and that can cause some headache to track/debug.

**Reply by stephentoub** [reply]
> while HttpClient.Shared can be re-configured in different places and that can cause some headache to track/debug

Most (all?) of the state mutation on HttpClient throws if you try to change it once the first request is made. I expect if such a Shared were exposed, we'd just set that bit from the get-go, and there wouldn't be such concerns.

I expect such a Shared property would be useful for very basic scenarios. But it wouldn't be usable if you needed to configure anything on the client or underlying handler, and the need to do so is very common.

**Reply by Xyncgas** [reply]
I agree
